YvesIraguha/personalFinanceMobile
# #172122189 Replace Redux with Apollo

#### What does this PR do?
- replace redux implementation with apollo 
#### Description of Task to be completed?
- replace fetch expenses with useQuery hook
- replace record expenses with useMutation hook
- replace edit expense with useMutation hook
- replace delete expense with useMutation hook
- enable optimisticUI
- refactor expense creation
- refactor login screen to use apollo-client
- remove redux files


#### How should this be manually tested?
- clone the repo and checkout the branch `ch-replace-redux-with-apollo-#172122189`
- Go through the codebase, you should see that now the codebase uses apollo-client instead of redux. 
#### Any background context you want to provide?
- in order to implement offline capability, the codebase had to be converted to use apollo-client instead of redux because it consumes a graphql API. 
#### What are the relevant pivotal tracker stories?
#172122189
